src/transformStateWithClones.js

```diff
@@ -6,8 +6,35 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can make a destructuring of all needed props of action to make your code cleaner